Release pipeline fails on darwin/arm64 cross-compile

## Problem

Release workflow has failed for the last two tags (v0.25.0, v0.25.1). Last published release is v0.24.0. GoReleaser fails during the `building binaries` phase:

```
build failed: exit status 1: # github.com/zmap/zcrypto/ct/x509
.../zmap/zcrypto@v0.0.0-20250618174828-7ca6a82cf2d4/ct/x509/root.go:21:32: undefined: loadSystemRoots
target=darwin_arm64_v8.0
```

## Root cause

`zmap/zcrypto/ct/x509` (pulled in transitively via `zgrab2`, added in v0.25.0 for banner enrichment) has platform-gated root-cert files whose build tags leave `darwin/arm64 + CGO_ENABLED=0` uncovered:

| File | Build tag |
|---|---|
| `root_cgo_darwin.go` | `cgo && !arm && !arm64 && !ios` |
| `root_nocgo_darwin.go` | `!arm && !arm64 && !ios` |
| `root_darwin_armx.go` | `cgo && darwin && (arm \|\| arm64 \|\| ios)` |

Intersection for `darwin/arm64 && !cgo` is empty — no file defines `loadSystemRoots`. This exists in the latest upstream zcrypto too (checked `v0.0.0-20260413215825-aacf0e34cc16`).

`.goreleaser.yml` sets `CGO_ENABLED=0` globally (needed so linux binaries are static for minimal Docker images), which is why the failure only hits on darwin/arm64 cross-compile.

## Fix (this issue)

Drop darwin from the goreleaser build matrix for now. Mac users can `go install ./cmd/scanorama` locally where CGO defaults to on and `root_darwin_armx.go` resolves `loadSystemRoots`.

## Follow-ups (separate issues)

- Restore darwin binaries via a macos-latest matrix job or `goreleaser-cross` with CGO enabled for darwin targets
- `release.yml` `needs: ci` is satisfied when path filters in `main.yml` skip all real test jobs on tag pushes — gate hole that masked this failure from surfacing earlier

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -5,11 +5,13 @@ before:
     - go mod tidy
 
 builds:
+  # darwin builds are disabled until the zgrab2 → zmap/zcrypto/ct/x509
+  # no-cgo darwin/arm64 gap is resolved (missing loadSystemRoots). Mac users
+  # can `go install ./cmd/scanorama` locally where CGO defaults to on.
   - main: ./cmd/scanorama
     binary: scanorama
     goos:
       - linux
-      - darwin
     goarch:
       - amd64
       - arm64

